### ISSUE TITLE
Consul intermittent hang on restart when using PVs

### ISSUE THREAD

**Issue by IBMRob** [hashicorp/consul#4239]
#### Overview of the Issue

Running Consul in Kubenetes we are hitting a problem where by sometimes consul is failing to startup. It passes the initial ping checks but then hangs before starting.

#### Reproduction Steps

Not easy to replicate as it appears to happen intermittently. 
Consul deployed (using public helm charts with slight modification to go across AZs) with 6 instances all with persistent volumes in the IBM Cloud.
When we have worker failures and consul restarts it occasionally will hang.

It appears that if we delete everything on the persistent volume and then kill the pod then it will startup cleanly.

### Consul info for both Client and Server

<details>
  <summary>Server info</summary>

```
/ # consul info
agent:
	check_monitors = 0
	check_ttls = 0
	checks = 0
	services = 0
build:
	prerelease = 
	revision = 9a494b5f
	version = 1.0.6
consul:
	bootstrap = false
	known_datacenters = 1
	leader = true
	leader_addr = 172.30.179.148:8300
	server = true
raft:
	applied_index = 1754066
	commit_index = 1754066
	fsm_pending = 0
	last_contact = 0
	last_log_index = 1754066
	last_log_term = 169
	last_snapshot_index = 1753320
	last_snapshot_term = 169
	latest_configuration = [{Suffrage:Voter ID:18e34231-ff93-8669-270b-aea1af2305e4 Address:172.30.71.173:8300} {Suffrage:Voter ID:f115ba5c-4eaf-0a24-9281-22f8e7e36c9b Address:172.30.250.149:8300} {Suffrage:Voter ID:4efb9d09-4b0b-b2b3-f233-72a769451e2f Address:172.30.69.140:8300} {Suffrage:Voter ID:62ff2553-18c3-c835-666f-41a48738a915 Address:172.30.179.148:8300}]
	latest_configuration_index = 1731740
	num_peers = 3
	protocol_version = 3
	protocol_version_max = 3
	protocol_version_min = 0
	snapshot_version_max = 1
	snapshot_version_min = 0
	state = Leader
	term = 169
runtime:
	arch = amd64
	cpu_count = 4
	goroutines = 305
	max_procs = 4
	os = linux
	version = go1.9.3
serf_lan:
	coordinate_resets = 0
	encrypted = true
	event_queue = 0
	event_time = 56
	failed = 280
	health_score = 0
	intent_queue = 0
	left = 311
	member_time = 134201
	members = 626
	query_queue = 0
	query_time = 7
serf_wan:
	coordinate_resets = 0
	encrypted = true
	event_queue = 0
	event_time = 1
	failed = 2
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 75224
	members = 6
	query_queue = 0
	query_time = 1
```

</details>

### Operating system and Environment details

Running on IBM Cloud.

### Log Fragments

Logs of one of the servers which is failing to startup.
```
Waiting for 3 servers to be up
Checking for consul-consul-az1-0.consul-consul.default.svc to come up
consul-consul-az1-0.consul-consul.default.svc is up
Checking for consul-consul-az1-1.consul-consul.default.svc to come up
consul-consul-az1-1.consul-consul.default.svc is up
Checking for consul-consul-az2-0.consul-consul.default.svc to come up
consul-consul-az2-0.consul-consul.default.svc is up
Checking for consul-consul-az2-1.consul-consul.default.svc to come up
consul-consul-az2-1.consul-consul.default.svc is up
Checking for consul-consul-az3-0.consul-consul.default.svc to come up
consul-consul-az3-0.consul-consul.default.svc is up
Checking for consul-consul-az3-1.consul-consul.default.svc to come up
consul-consul-az3-1.consul-consul.default.svc is up
bootstrap_expect > 0: expecting 3 servers
==> Starting Consul agent...
```

Use `-log-level=TRACE` on the client and server to capture the maximum log detail.

**Comment by pearkes** [comment]
Can you provide your configuration files for the Consul servers (I believe this would be defined in Helm in your scenario)?

**Comment by pearkes** [comment]
Given we haven't heard anything based on our suggestions/questions above I'm going to close this issue, but I encourage you to comment and we can re-open it if you want to pick this up again.

Alternatively, if things have changed dramatically, feel free to create a new issue or PR.
